third_party/pcsc-lite/naclport/server/src/readerfactory_nacl.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016 Google Inc.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+// 3. The name of the author may not be used to endorse or promote products
+//    derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+// OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+// IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+// NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+// THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// This file contains a replacement function for the original readerfactory.c
+// PC/SC-Lite internal implementation.
+
+#include <ppapi/cpp/instance.h>
+#include <ppapi/cpp/var.h>
+#include <ppapi/cpp/var_dictionary.h>
+
+#include <google_smart_card_common/pp_var_utils/construction.h>
+#include <google_smart_card_pcsc_lite_server/global.h>
+
+extern "C" {
+#include "readerfactory.h"
+
+LONG RFAddReaderOriginal(const char *readerNameLong, int port,
+    const char *library, const char *device);
+LONG RFRemoveReaderOriginal(const char *readerName, int port);
+}
+
+namespace {
+
+const char kTypeMessageKey[] = "type";
+const char kDataMessageKey[] = "data";
+const char kReaderInitAddMessageType[] = "reader_init_add";
+const char kReaderFinishAddMessageType[] = "reader_finish_add";
+const char kReaderRemoveMessageType[] = "reader_remove";
+const char kNameMessageKey[] = "readerName";
+const char kPortMessageKey[] = "port";
+const char kDeviceMessageKey[] = "device";
+const char kReturnCodeMessageKey[] = "returnCode";
+
+void post_message(const char* type, const pp::VarDictionary& message_data)
+{
+  pp::VarDictionary message;
+  message.Set(kTypeMessageKey, type);
```

Comment:
It's better to use VarDictBuilder all the time, because it adds some assertions (first, because pp::VarDictionary::Set may return error; second, because trying to overwrite a key is usually an error), and because it ensures the proper value conversion (though in this exact case the standard pp::VarDictionary API is enough).